modules/ducktests/tests/ignitetest/tests/cellular_affinity_test.py

```diff
@@ -0,0 +1,123 @@
+                                <property name="affinityBackupFilter">
+                                    <bean class="org.apache.ignite.internal.ducktest.tests.cellular_affinity_test.CellularAffinityBackupFilter">
+                                        <constructor-arg value="{{ attr }}"/>
+                                    </bean>
+                                </property>
+                            </bean>
+                        </property>
+                        <property name="name" value="{{ cacheName }}"/>
+                        <property name="backups" value="{{ nodes }}"/> 
+                    </bean>
+                </list>
+            </property>
+        """
+
+    @staticmethod
+    def properties():
+        """
+        :return: Configuration properties.
+        """
+        return Template(CellularAffinity.CONFIG_TEMPLATE) \
+            .render(nodes=CellularAffinity.NUM_NODES,  # bigger than cell capacity (to handle single cell useless test)
+                    attr=CellularAffinity.ATTRIBUTE,
+                    cacheName=CellularAffinity.CACHE_NAME)
+
+    def __init__(self, test_context):
+        super(CellularAffinity, self).__init__(test_context=test_context)
+
+    def setUp(self):
+        pass
+
+    def teardown(self):
+        pass
+
+    @cluster(num_nodes=NUM_NODES * 3 + 1)
+    @parametrize(version=str(DEV_BRANCH))
+    def test(self, version):
+        """
+        Test Cellular Affinity scenario (partition distribution).
+        """
+        ignite_version = IgniteVersion(version)
+
+        ignites1 = IgniteService(
+            self.test_context,
+            num_nodes=CellularAffinity.NUM_NODES,
+            version=ignite_version,
+            properties=self.properties(),
+            jvm_opts=['-D' + CellularAffinity.ATTRIBUTE + '=1'])
+
+        ignites1.start()
+
+        ignites2 = IgniteService(
+            self.test_context,
+            num_nodes=CellularAffinity.NUM_NODES,
+            version=ignite_version,
+            properties=self.properties(),
+            jvm_opts=['-D' + CellularAffinity.ATTRIBUTE + '=2'])
+
+        ignites2.start()
+
+        ignites3 = IgniteService(
+            self.test_context,
+            num_nodes=CellularAffinity.NUM_NODES,
+            version=ignite_version,
+            properties=self.properties(),
+            jvm_opts=['-D' + CellularAffinity.ATTRIBUTE + '=XXX', '-DRANDOM=42'])
+
+        ignites3.start()
+
+        checker = IgniteApplicationService(
+            self.test_context,
+            java_class_name="org.apache.ignite.internal.ducktest.tests.cellular_affinity_test.DistributionChecker",
+            params={"cacheName": CellularAffinity.CACHE_NAME,
+                    "attr": CellularAffinity.ATTRIBUTE,
+                    "nodesPerCell": self.NUM_NODES},
+            version=ignite_version)
+
+        checker.start()
```

Comment:
Test without assertions or meauserements looks a little bit weird. May be assert smth in checker log? (log_monitor and wait_until, I mean)